src/book.py

```diff
@@ -205,6 +241,54 @@ def snippet(self, length, on, non_cont=False, inplace=False):
         else:
             return BookText(rawtext=snippet, author=self.author,
                             title=self.title, meta=self.meta)
+        
+    def random_snippet(self, on, n_groups=1, n_on=1, start_point=1, randomized = True, with_replacement=False, rem_stopwords=False, random_seed=0):
```

Comment:
Can this just be added as part of the snippet function? I thought we talked about adding features to the base method with an eye on expandability.

Reply:
@kdettman I commented on this in a substantial amount of detail in my review; you may want to read through my comments there and see if you agree/disagree

Reply:
I also see that this was addressed in the more general comment by Elaad above.